File: CS/WindowsFormsApplication9/CustomDateEditCalendar.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using DevExpress.XtraEditors.Controls;
using DevExpress.XtraEditors.Repository;
using DevExpress.XtraEditors.ViewInfo;

namespace WindowsFormsApplication9 {
    public class CustomDateEditCalendar : DateEditCalendar {
        public CustomDateEditCalendar(RepositoryItemDateEdit item, object editDate)
            : base(item, editDate) {

        }
        protected override DateEditInfoArgs CreateInfoArgs() {
            return new CustomDateEditInfoArgs(this);
        }
    }
    
}

File: VB/WindowsFormsApplication9/CustomDateEditCalendar.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports DevExpress.XtraEditors.Controls
Imports DevExpress.XtraEditors.Repository
Imports DevExpress.XtraEditors.ViewInfo

Namespace WindowsFormsApplication9
    Public Class CustomDateEditCalendar
        Inherits DateEditCalendar

        Public Sub New(ByVal item As RepositoryItemDateEdit, ByVal editDate As Object)
            MyBase.New(item, editDate)

        End Sub
        Protected Overrides Function CreateInfoArgs() As DateEditInfoArgs
            Return New CustomDateEditInfoArgs(Me)
        End Function
    End Class

End Namespace
